### PR TITLE
Bugfix spack ubuntu CPU build

### PR DESCRIPTION
**Pull request by cameronrutherford**
Adds `--fail-fast` to the spack install to avoid the installation running for any longer than it needs to

### CODE REVIEW

**Comment by cameronrutherford** [comment]
Noticed the ubuntu build was failing in #131, so this is an attempt to speed up the rate of failure in those builds....

**Comment by cameronrutherford** [comment]
Not really sure why this build is now failing... Ideas:

- [ ] Use a spack provided base image
- [ ] Pin spack version
- [ ] Re-use packages more aggressively

Still really unclear why this is happening since it's an error in getting the C compiler to even do anything

File: .github/workflows/spack_cpu_build.yaml [hunk 1]
```diff
@@ -116,31 +120,38 @@ jobs:
...
+              build_stage: $SPACK_CACHE/build_stage
               install_tree:
                 root: /opt/spack
                 padded_length: False
             mirrors:
               local-buildcache: oci://${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
               spack: https://binaries.spack.io/develop
+            packages:
+              all:
+                require: "%gcc"
```
Comment (cameronrutherford) [inline .github/workflows/spack_cpu_build.yaml:135]: For some reason, clang was getting picked up by default as the compiler, and this caused spack to fail horribly. This seems to be all that was necessary to fix the issue

**Comment by cameronrutherford** [comment]
Ready for review + merge after pipelines pass. Since this uses a newer gcc and ubuntu version, the whole build is happening from scratch this first time. Future runs should be quicker

**Comment by cameronrutherford** [comment]
All green check marks, so I will merge once review is finished.

**Review by jaelynlitz — APPROVED** [review]

**Pull request merged**